bson/src/main/scala/types.scala

```diff
@@ -996,7 +996,10 @@ case class BSONDocument(stream: Stream[Try[BSONElement]])
    */
   def getAsTry[T](key: String)(implicit reader: BSONReader[_ <: BSONValue, T]): Try[T] = {
     val tt = getTry(key)
-    tt.flatMap { element => Try(reader.asInstanceOf[BSONReader[BSONValue, T]].read(element)) }
+    tt.flatMap { element: BSONValue =>
```

Comment:
```suggestion
    tt.flatMap {
      case BSONNull =>
        Failure(DocumentKeyIsNull(key))

      case element => 
        Try(reader.asInstanceOf[BSONReader[BSONValue, T]].read(element))
    }
```

Reply:
I used recover to allow the (unlikely) case of a `reader` which actually does something with `null`. This may be far fetched (I never created such a reader myself) so returning direcly an error is probably ok.

Reply:
People wanting to deal with raw value could do it without reader. I would keep it the simpler.

FYI the [BSON refactoring](https://github.com/cchantep/RM-bson/blob/master/api/src/main/scala/types.scala) avoid such dirty situation.